dvc/config_schema.py

```diff
@@ -97,6 +97,9 @@ class RelPath(str):
     "ask_password": Bool,
     "ssl_verify": Any(Bool, str),
     "method": str,
+    "connect_timeout": All(Coerce(float), Range(0, min_included=True)),
+    "sock_connect_timeout": All(Coerce(float), Range(0, min_included=True)),
+    "sock_read_timeout": All(Coerce(float), Range(0, min_included=True)),
```

Comment:
Wdyt of having one single `timeout` config?

Reply:
The fact that we are using `aiohttp` is an implementation detail (we don't even use `aiohttp` directly, we use `HTTPFilesystem`). I think having a simple `timeout` is simpler and translates easily between different implementations in the future.

Reply:
I thought about it for a while. I had only one at first, although it felt odd to set all of these values to the same value: for example setting `connect_timeout` to a large value (or disabling it), it might end up hanging for a while (or forever) for something that might fail immediately. I hate the complexity of having three different values (and exposing all of these confusing implementation details).

A tradeoff might be only exposing connect timeout and sock_read timeout (merging the two different connect timeouts into a single one). What do you think?

Reply:
In both [`requests`][1]/[`httpx`][2]/[`urllib3`][3]/[`urllib.request`][4], a single `timeout` value is applied to both `connect` and `read` timeout.

And, is there a difference between `connect_timeout` and `sock_connect_timeout` from a user perspective? Reading through the [aiohttp's docs][5], it seems the former one is when the connection is from a pool, and the other is when it's not from a pool.

> `connect`
>
> The maximal number of seconds for connection establishment of a new connection or for waiting for a free connection from a pool if pool connection limits are exceeded.
>
> `sock_connect`
>
> The maximal number of seconds for connecting to a peer for a new connection, not given from a pool.


[1]: https://requests.readthedocs.io/en/latest/user/advanced/#timeouts
[2]: https://www.python-httpx.org/advanced/#setting-a-default-timeout-on-a-client
[3]: https://github.com/urllib3/urllib3/blob/1a897ddfe2f7e017a354ec99b924b5d0afc166d8/src/urllib3/util/timeout.py#L190
[4]: https://docs.python.org/3/library/urllib.request.html#urllib.request.urlopen
[5]: https://docs.aiohttp.org/en/stable/client_quickstart.html#timeouts

Reply:
> In both requests/httpx/urllib3/urllib.request, a single timeout value is applied to both connect and read timeout.

True, although `httpx` does allow for granular timeout configuration (connect/read/write/pool), `urllib3` does allow for  granular read/write timeouts.

If we want to compare with other filesystems we support, such as S3, in that case we do provide some granularity for timeouts:
https://github.com/iterative/dvc/blob/3bbf8fa7d2c517af6b98e2d26f44c2618398e248/dvc/config_schema.py#L165-L166

these seem to reflect 1:1  the underlying S3Fs/botocore config options.

Regarding `connect` and `sock_connect`, it may be enough to only expose `sock_connect`, it would also match the S3 config schema. For other filesystems:
- Webdav uses a single timeout value, passed straight to httpx's timeout
- ssh has a single timeout value

Reply:
@skshetry any thoughs? I'd merge it as is, though if you are concerned with the possibilty of switching to other implementations in the future, I can just get rid of `connect_timeout` and just keep `sock_connect_timeout`) and `sock_read`, as it seems that most libraries provide these two values (or maybe just set `connect_timeout` and `sock_connect_timeout` to the same value).

Reply:
https://github.com/iterative/dvc-http/pull/29